### PR TITLE
Feature/add agent search

### PR DESCRIPTION
**Pull request by emrgnt-cmplxty**
# Description

This framework facilitates seamless integration with the AgentSearch dataset or hosted search APIs (e.g. Search Engines) and with RAG-specialized LLM's (e.g. Search Agents).

During query-time, the user passes in the query string, search provider (`bing`, `agent-search`), and RAG provider model (`SciPhi/Sensei-7B-V1`). 

To learn more, please refer to the documentation [here](https://agent-search.readthedocs.io/en/latest/).

## Type of Change

Please delete options that are not relevant.
- [x] New Loader/Tool

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally by replacing example in `download_loader("AgentSearch")` in base.py with `AgentSearchReader`

### CODE REVIEW

File: llama_hub/agent_search/base.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            "`agent-search` package not found, please run `pip install agent-search`"
+        )
+        try:
+            import agent_search  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        from agent_search import SciPhi
+
+        self._client = SciPhi(api_base=api_base, api_key=api_key)
```
Comment (logan-markewich) [inline llama_hub/agent_search/base.py:56]: Can the API key also be an env var? Does the api_base default to something internally?

I noticed in the README that the API key and API base was not mentioned

File: llama_hub/library.json [hunk 2]
```diff
@@ -1,4 +1,8 @@
 {
+  "AgentSearch": {
```
Comment (logan-markewich) [inline llama_hub/library.json:2]: I think you need to rename `AgentSearch` here to match the class name -- `AgentSearchReader`

Comment (emrgnt-cmplxty) [reply llama_hub/library.json:2]: got it, ty

File: llama_hub/agent_search/base.py (hunk 1 above)
Comment (emrgnt-cmplxty) [reply llama_hub/agent_search/base.py:56]: SciPhi defaults to SCIPHI_API_KEY if no api_key is provided, this just gives the user additional optionality

**Comment by logan-markewich** [comment]
@emrgnt-cmplxty if you are cool with the changes/additions I made, I will merge away :)

**Pull request merged**